src/ArmTemplates/Common/Templates/Abstractions/TemplateParameterProperties.cs

```diff
@@ -1,11 +1,22 @@
-﻿namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Templates.Abstractions
+﻿using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Commands.Applications;
+
+namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Templates.Abstractions
 {
     public class TemplateParameterProperties
     {
-        public string type { get; set; }
-        public TemplateParameterMetadata metadata { get; set; }
-        public string[] allowedValues { get; set; }
-        public string defaultValue { get; set; }
-        public string value { get; set; }
+        public string Type { get; set; }
+        public TemplateParameterMetadata Metadata { get; set; }
+        public string[] AllowedValues { get; set; }
+        public string DefaultValue { get; set; }
+        public string Value { get; set; }
+
+        public static TemplateParameterProperties Create(string metadataDescription, string type) => new()
+        {
+            Metadata = new()
```

Comment:
Nit: I'm not a fan of the `new()` in cases like this. Yes it's shorter but if you want to know what it's actually creating you have to look at the type of what you're assigning it to, which wastes time.

Reply:
Made it explicit. Fair enough about your codestyle :)
I am just trying some c# new features and try to come up to some ideology of code writing, so thanks for providing your thoughts